src/libraries/Microsoft.Extensions.DependencyModel/src/Utf8JsonReaderExtensions.cs

```diff
@@ -14,19 +15,19 @@ internal static class Utf8JsonReaderExtensions
         public static bool IsTokenTypeProperty(this ref Utf8JsonReader reader)
             => reader.TokenType == JsonTokenType.PropertyName;
 
-        public static bool TryReadStringProperty(this ref Utf8JsonReader reader, out string name, out string value)
+        public static bool TryReadStringProperty(this ref Utf8JsonReader reader, [MaybeNullWhen(false)] out string name, out string? value)
         {
             name = null;
             value = null;
             if (reader.Read() && reader.IsTokenTypeProperty())
             {
-                name = reader.GetString();
+                name = reader.GetString()!;
 
                 if (reader.Read())
                 {
                     if (reader.TokenType == JsonTokenType.String)
                     {
-                        value = reader.GetString();
+                        value = reader.GetString()!;
```

Comment:
Why is this `!` necessary?

Reply:
It's not, but as I said in PRs description, it looks like `reader.GetString()` is always assumed to be non-null.
I'll update annotations to reflect that it can return null.